### PR TITLE
fix: 💄 Layout.Sider collapsedWidth cannot work without @menu-collapsed-width

### LINKED ISSUE #26816: 想了解一下 Menu 组件收起和展开时带有子菜单的菜单项和无子菜单的菜单项的动画效果为何不同？
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/menu-cn/#components-menu-demo-inline-collapsed](https://ant.design/components/menu-cn/#components-menu-demo-inline-collapsed)

### Steps to reproduce
点击收起展开

### What is expected?
希望收起和展开时所有的菜单项走统一的动画。

### What is actually happening?
收起的时候包含子菜单的菜单项会直接消失，而无子菜单项的菜单项会有一个向左渐隐的过程。展开时候动效取反。

| Environment | Info |
|---|---|
| antd | 4.6.4 |
| React | 16.8.6 |
| System | mac os 10.14.1 |
| Browser | chrome 73.0.3683.103 |

---
如果是刻意为之，想知道这个交互动效是想要表达什么？包含子菜单项的右侧剪头已经足够表达其可以被二次折叠展开了，这里的区别动画貌似有些多余，反而造成动效不统一，也不太符合：https://ant.design/docs/spec/motion-cn#%E5%85%8B%E5%88%B6 这里的克制原则。

### LINKED ISSUE #26992: collapsedWidth is broken
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-sejf5?file=/index.js)

### Steps to reproduce
set collapsedWidth to 50

### What is expected?
collapsedWidth should change all items inside sider

### What is actually happening?
image

collapsedWidth isn't changing all items inside sider

I saw this issue https://github.com/ant-design/ant-design/issues/11422 and your answer @afc163 

but I think the library should do it, I don't have override less file (I'm using create react app and I don't want to add 2 npm packages just for it), or at least specify it in the docs.

thanks 🤍

| Environment | Info |
|---|---|
| antd | 4.6.6 |
| React | 16.13.0 |
| System | windows 10 |
| Browser | chrome 85 |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #26992 
close #26816

### 💡 Background and solution

Now we can use Layout.Sider `collapsedWidth` without modify less variable `@menu-collapsed-width`.

Fixed Menu collapsed animation by updating dom structure in https://github.com/react-component/menu/pull/343

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Layout.Sider `collapsedWidth` cannot work without  modifying`@menu-collapsed-width`.<br />Fix Menu inline collapsed broken animation. |
| 🇨🇳 Chinese | 修复 Layout.Sider 指定 `collapsedWidth` 后侧边菜单部分宽度不对的问题。<br />修复 Menu 收缩侧边菜单时动画不统一问题。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/menu/__tests__/__snapshots__/demo.test.js.snap [hunk 1]
```diff
@@ -565,6 +571,12 @@ exports[`renders ./components/menu/demo/inline-collapsed.md correctly 1`] = `
           class="ant-menu-submenu-arrow"
         />
       </div>
+      <div>
+        <div
+          class="ant-menu-submenu ant-menu-submenu-popup"
+          style="opacity:0;pointer-events:none"
+        />
+      </div>
```
Comment (afc163) [inline components/menu/__tests__/__snapshots__/demo.test.js.snap:579]: 这个 dom 结构很奇怪，还需要排查一下。

File: components/menu/style/rtl.less [hunk 2]
```diff
@@ -136,7 +136,7 @@
   &-inline-collapsed&-vertical {
     .@{menu-prefix-cls}-submenu-title {
       .@{menu-prefix-cls}-rtl& {
-        padding: 0 (@menu-collapsed-width - @menu-icon-size-lg) / 2;
+        padding: 0 ~'calc(50% - @{menu-icon-size-lg} / 2)';
```
Comment (afc163) [inline components/menu/style/rtl.less:139]: collapsedWidth 之前需要配合 @menu-collapsed-width 的原因是需要配合切换动画，所有 icon 居中需要用 padding 来做（text-align: center 走不了 transition）。现在 padding 换了一个取巧的方式实现居中效果，脱离了对 @menu-collapsed-width 的依赖。

Comment (hengkx) [reply components/menu/style/rtl.less:139]: 为啥  我看了 有一种 说不出的 奇怪😱

Comment (xrkffgg) [reply components/menu/style/rtl.less:139]: 顺滑了，这玩意不能一直盯着

Comment (chinamcafee) [reply components/menu/style/rtl.less:139]: antd-pro的Menu控件也不没有动画效果，请问要怎么升级这个修复呢？

**Comment by xrkffgg** [comment]
pr-triage  神奇的修改了 label

**Pull request merged**